src/GafferArnoldUI/BarndoorVisualiser.cpp

```diff
@@ -0,0 +1,261 @@
+
+#include "IECore/MeshPrimitive.h"
+#include "IECoreGL/CurvesPrimitive.h"
+#include "IECore/Shader.h"
+#include "IECoreGL/Group.h"
+#include "IECoreGL/ToGLMeshConverter.h"
+#include "IECoreGL/ShaderStateComponent.h"
+#include "IECoreGL/ShaderLoader.h"
+#include "IECoreGL/TextureLoader.h"
+#include "IECore/ObjectVector.h"
+#include "IECoreGL/Renderable.h"
+
+#include "Gaffer/Metadata.h"
+#include "GafferSceneUI/StandardLightVisualiser.h"
+#include "GafferSceneUI/LightFilterVisualiser.h"
+
+using namespace Imath;
+using namespace IECore;
+using namespace IECoreGL;
+using namespace GafferSceneUI;
+
+namespace
+{
+
+enum BarndoorLocation { TOP, RIGHT, LEFT, BOTTOM };
+
+float parameterOrDefault( const IECore::CompoundData *data, const char *key, const float def )
+{
+	ConstFloatDataPtr member = data->member<const FloatData>( key );
+	if( member )
+	{
+		return member->readable();
+	}
+	return def;
+}
+
+const char *barndoorFragSource()
+{
+	return
+		"#if __VERSION__ <= 120\n"
+		"#define in varying\n"
+		"#endif\n"
+		""
+		"void main()"
+		"{"
+			"if( (int(gl_FragCoord.x) + int(gl_FragCoord.y)) % 2 == 0 )"
+			"{"
+				"discard;"
+			"}"
+			"else"
+			"{"
+				"gl_FragColor = vec4(0, 0, 0, 1);"
+			"}"
+		"}"
+		;
+}
+
+void addBarndoor( IECoreGL::GroupPtr result, BarndoorLocation location, float cornerLeft, float cornerRight )
+{
+
+	if( !( cornerLeft > 0 || cornerRight > 0 ) )
+	{
+		return;
+	}
+
+	cornerLeft = 1 - cornerLeft * 2.0;
+	cornerRight = 1 - cornerRight * 2.0;
+
+	IntVectorDataPtr vertsPerPoly = new IntVectorData;
+	std::vector<int> &vertsPerPolyVec = vertsPerPoly->writable();
+	vertsPerPolyVec.push_back( 4 );
+
+	IntVectorDataPtr vertIds = new IntVectorData;
+	std::vector<int> &vertIdsVec = vertIds->writable();
+	vertIdsVec.push_back( 0 );
+	vertIdsVec.push_back( 1 );
+	vertIdsVec.push_back( 2 );
+	vertIdsVec.push_back( 3 );
+
+	V3fVectorDataPtr p = new V3fVectorData;
+	std::vector<V3f> &pVec = p->writable();
+	pVec.push_back( V3f( -1, 1, 0  ) );
+	pVec.push_back( V3f( 1, 1, 0  ) );
+	pVec.push_back( V3f( 1, cornerRight, 0  ) );
+	pVec.push_back( V3f( -1, cornerLeft, 0  ) );
+
+	IECore::MeshPrimitivePtr mesh = new IECore::MeshPrimitive( vertsPerPoly, vertIds, "linear", p );
+
+	Imath::M44f trans;
+	switch( location )
+	{
+	case TOP:
+		break;
+	case BOTTOM:
+		trans.rotate( V3f( 0, 0, M_PI ) );
+		break;
+	case LEFT:
+		trans.rotate( V3f( 0, 0, M_PI / 2.0 ) );
+		break;
+	case RIGHT:
+		trans.rotate( V3f( 0, 0, M_PI / -2.0 ) );
+		break;
+	}
+
+	ToGLMeshConverterPtr meshConverter = new ToGLMeshConverter( mesh );
+
+	IECoreGL::GroupPtr barndoorGroup = new IECoreGL::Group();
+	barndoorGroup->getState()->add( new IECoreGL::Primitive::Selectable( false ) );
+	barndoorGroup->addChild( IECore::runTimeCast<IECoreGL::Renderable>( meshConverter->convert() ) );
+	barndoorGroup->setTransform( trans );
+
+	result->addChild( barndoorGroup );
+}
+
+} // namespace
+
+namespace GafferArnoldUI
```

Comment:
I think everything can actually be in the anonymous namespace for all of these visualisers.